packages/subproviders/src/subproviders/ledger.ts

```diff
@@ -130,7 +102,10 @@ export class LedgerSubprovider extends BaseWalletSubprovider {
      */
     public async signTransactionAsync(txParams: PartialTxParams): Promise<string> {
         LedgerSubprovider._validateTxParams(txParams);
-        this._ledgerClientIfExists = await this._createLedgerClientAsync();
+        const initialDerivedKey = await this._initialDerivedKeyAsync();
```

Comment:
Why not set the initialDerivedKey in the constructor? We only need to compute this at instantiation, and if the `baseDerivationPath` is updated, no?

Reply:
I think I prefer it to be lazy, if we do it in `setPath` then it becomes async and could potentially throw (from a ledger error). Whereas this will throw when the user is trying to do an action and the developer is ready for any errors.